### ISSUE TITLE
Distroless CBL-Mariner images contain RPM database

### ISSUE THREAD

**Issue by mthalman** [dotnet/dotnet-docker#3941]
There's about 500KB of unnecessary files in the distroless CBL-Mariner images as part of the RPM database. These are located at the path /var/lib/rpm:

![image](https://user-images.githubusercontent.com/15789599/180498642-cc08f548-dd77-4c5f-ac0b-8a383cb1e3c3.png)

While not needed by an application being hosted in the container, the RPM DB does provide some usefulness when it comes to scanning tools. That's actually how I came to realize that RPM DB was even in the container. I saw that the SBOM generated for the mcr.microsoft.com/dotnet/runtime-deps:6.0-cbl-mariner1.0-distroless image correctly listed its installed packages. Which didn't make sense to me since there shouldn't have existed a way to enumerate what packages were installed. The SBOM generation makes use of [syft](https://github.com/anchore/syft) which scans the RPM DB: https://github.dev/anchore/syft/blob/b720a3c81c677820c69b5652ee8cb338731d4b47/syft/pkg/rpmdb_metadata.go#L17. This led to me the existence of the populated RPM DB in the container. So it is actually useful to have it.

Before we just go ahead and delete this from the images, we need to do some investigation to determine if there may be other reasons the existence of this DB can be useful. Can it be used as part of a scenario for authors of downstream Dockerfiles that need to install additional packages? That may help correctly identifying the existence of already installed package dependencies.

If we do decide to delete this from the image, what is the story for describing what packages are installed for scanning tools? [CBL-Mariner has defined a manifest file](https://github.com/microsoft/CBL-Mariner/blob/2.0/toolkit/docs/how_it_works/5_misc.md#rpm-manifest-file) which is intended to fill this role but there's no tooling story for downstream image consumers on updating this.

Bottom line is that we need a better picture of the overall developer and tooling UX for working with distroless CBL-Mariner images.

**Comment by eric-desrochers** [comment]
Adding @mandeepsplaha who is our Mariner container SME to assist Matt with this situation.

**Comment by mthalman** [comment]
[Triage]
Two scenarios need to be looked at:

* Container scanning tools should be able to accurately determine which packages are installed.

* RPM installation should be able to determine which packages are installed in order to avoid duplicate installation of a package that already exists from another image layer. This is specific to the Dockerfile pattern shown below that illustrates how to install additional packages in a distroless Mariner container image:

```Dockerfile
FROM mcr.microsoft.com/cbl-mariner/base/core:2.0 AS installer

RUN mkdir /staging \
    && tdnf install -y --releasever=2.0 --installroot /staging <package-name>

# Clean up staging
RUN rm -rf /staging/etc/dnf \
    && rm -rf /staging/run/* \
    && rm -rf /staging/var/cache/dnf \
    && find /staging/var/log -type f -size +0 -delete


FROM mcr.microsoft.com/dotnet/aspnet:6.0-cbl-mariner2.0-distroless AS base


FROM base

COPY --from=installer /staging/ /

# Ensures that the contents of the staging directory doesn't overwrite the default users/groups from the base image
COPY --from=base /etc/passwd /etc/passwd
COPY --from=base /etc/group /etc/group
```

The meta point here is that tools used in container scenarios should have the ability to read a single package "DB" that's contained in the image. This may involve needing to coordinate with Mariner to have a solution.

**Comment by mthalman** [comment]
I've logged a feature request for Mariner to provide a tooling story around package installation for distroless scenarios: https://microsoft.visualstudio.com/OS/_workitems/edit/40762920. That includes both  scenarios [described above](https://github.com/dotnet/dotnet-docker/issues/3941#issuecomment-1197188181).

To move forward with this particular issue, we'll proceed with getting the RPM database deleted.
